apps/aecore/src/aec_dev_reward.erl

```diff
@@ -0,0 +1,113 @@
+            AllocShares = lists:sum(Shares),
+            if AllocShares == 0 ->
+                    exit({invalid_protocol_beneficiaries, sum_shares_is_zero});
+               AllocShares > ?TOTAL_SHARES ->
+                    exit({invalid_protocol_beneficiaries, sum_shares_too_large});
+               AllocShares =< ?TOTAL_SHARES ->
+                    application:set_env(aecore, dev_reward_enabled, true),
+                    application:set_env(aecore, dev_reward_allocated_shares, AllocShares),
+                    application:set_env(aecore, dev_reward_beneficiaries, Beneficiaries)
+            end;
+        {error, Reason} ->
+            exit({invalid_protocol_beneficiaries, Reason})
+    end.
+
+cfg(Key, Default) ->
+    aeu_env:user_config([<<"chain">>, Key], Default).
+
+env(Key, Default) ->
+    aeu_env:get_env(aecore, Key, Default).
+
+enabled() ->
+    env(dev_reward_enabled, ?ENABLED).
+beneficiaries() ->
+    env(dev_reward_beneficiaries, ?BENEFICIARIES).
+total_shares() ->
+    ?TOTAL_SHARES.
+allocated_shares() ->
+    {ok, V} = aeu_env:get_env(aecore, dev_reward_allocated_shares),
+    V.
+
+activated(Height) ->
+    case env(dev_reward_activated, undefined) of %% for eunit to avoid mocking
+        undefined ->
+            aec_hard_forks:protocol_effective_at_height(Height) >= ?FORTUNA_PROTOCOL_VSN;
+        Val when is_boolean(Val) ->
+            Val
+    end.
+
+
+unparse_beneficiary({PK, Share}) ->
+    <<(aeser_api_encoder:encode(account_pubkey, PK))/binary, ":", (integer_to_binary(Share))/binary>>.
+
+parse_beneficiary(BeneficiaryShareStr) ->
+    Regex = "^(?'account'ak_[1-9A-HJ-NP-Za-km-z]*):(?'share'[0-9]+)$",
+    case re:split(BeneficiaryShareStr, Regex, [trim, group, {return, binary}]) of
+        [[<<>>, Account, Share]] ->
+            case aeser_api_encoder:safe_decode(account_pubkey, Account) of
+                {ok, PubKey} ->
+                    {PubKey, binary_to_integer(Share)};
+                {error, _} ->
+                    error({invalid_protocol_beneficiary, Account})
+            end;
+        _ ->
+            error({invalid_protocol_beneficiary_share, BeneficiaryShareStr})
+    end.
+
+parse_beneficiaries([_|_] = BeneficiarySharesStrs) ->
+    try lists:foldl(fun (Str, Acc) -> [parse_beneficiary(Str) | Acc] end,
+                    [], BeneficiarySharesStrs) of
+        Parsed ->
+            {ok, Parsed}
+    catch
+        error:Reason ->
+            {error, Reason}
+    end.
+
+
+split(BeneficiaryReward1, BeneficiaryReward2, NewestNodeHeight) ->
+    case enabled() andalso activated(NewestNodeHeight) of
+        true ->
+            AllocShares = allocated_shares(),
+            TotalShares = total_shares(),
+            AbsContrib1 = (BeneficiaryReward1 * AllocShares) div TotalShares,
+            AbsContrib2 = (BeneficiaryReward2 * AllocShares) div TotalShares,
+            DevContrib  = AbsContrib1 + AbsContrib2,
+            {Leftover, [{PK0, Amount0} | RemDevRewards]} =
```

Comment:
So this `PK0` is the first in the `Benefs0` var in `ensure_env()` - ok.

Reply:
Last in config. (foldl)